include/common/mir/executor.h

```diff
@@ -57,6 +57,47 @@ class NonBlockingExecutor: public Executor
 {
 };
 
+class ThreadPoolExecutor : public NonBlockingExecutor
```

Comment:
Ah! I only meant to move the `extern NonBlockingExecutor& thread_pool_executor;` into `mir/executors.h`.

That said, I don't think I *mind* that this is here.